server/internal/graphql/resolver/schema.resolvers.go

```diff
@@ -304,25 +306,110 @@ func (r *subscriptionResolver) SubscribeMeshSyncEvents(ctx context.Context, k8sc
 			continue
 		}
 		go func(ctxID string, brokerEventsChan chan *broker.Message) {
+			publishHandlerWithProcessing := processAndRateLimitTheResponseOnGqlChannel(resChan, 5*time.Second)
 			for event := range brokerEventsChan {
-				if event.EventType == broker.ErrorEvent {
+				if event.EventType == broker.ErrorEvent || isSubscriptionFlushed { // better close the parent channel, but it is throwing panic
 					// TODO: Handle errors accordingly
 					continue
 				}
+
 				// handle the events
 				res := &model.MeshSyncEvent{
 					ContextID: ctxID,
 					Type:      string(event.EventType),
 					Object:    event.Object,
 				}
-				resChan <- res
-				go r.Config.DashboardK8sResourcesChan.PublishDashboardK8sResources()
+				publishHandlerWithProcessing(res)
+				// go r.Config.DashboardK8sResourcesChan.PublishDashboardK8sResources()
```

Comment:
@theBeginner86 I didn't go deep in what this function is used for (since I have you 😉), but even without this publish endpoint, the dashboard data is being correctly synced and processed (as far as I could see). Would you help in checking and re-assuring, please educate me on this portion, I should have missed something.

Reply:
Thanks for pointing this out.

This `DashboardK8sResourcesChan.PublishDashboardK8sResources` is a function that publishes to the golang channels that are being used to track dashboard subscription responsible for dynamically updating `Workloads` widget in the dashboard.
If we remove this then those channels wouldn't receive updates when a new object is received from MeshSync and hence wouldn't be able to send updates in UI.

It has a sibling function [DashboardK8sResourcesChan.SubscribeDashboardK8sResources](https://github.com/meshery/meshery/blob/master/server/models/dashboard_k8s_resources.go#L20). This [subscribeClusterResources](https://github.com/meshery/meshery/blob/master/server/internal/graphql/resolver/kubernetes.go#L119) is invoked when UI sends a subscribe request to backend.

Even without this function, we would see data in Workload widget in dashboard because of [GetClusterResources](https://github.com/meshery/meshery/blob/master/server/internal/graphql/resolver/schema.resolvers.go#L151) query but that would just be a one time fetch and UI would update only on reload.

Reply:
Thanks @theBeginner86